pages/api/users/[username]/index.js

```diff
@@ -149,7 +160,14 @@ export default async function handler(req, res) {
   await getLocation(username, latestProfile);
   const profileWithLocation = await Profile.findOne({ username });
 
-  return res
-    .status(200)
-    .json({ username, ...data, location: profileWithLocation.location });
+  const retData = {
+    status: 200,
+    userData: {
+      username,
+      ...data,
+      location: profileWithLocation.location
+    }
+  };
+  
+  return JSON.parse(JSON.stringify(retData));
```

Comment:
Why do we need to do both?

Reply:
The reason for this is to return an object that can be translated to JSON.
If I don't do this, Next complains when I try to include this object as part of the data returned from `getServerSideProps`.
I think it has something to do with converting Date structures to something JSON can handle.

Reply:
oh I see, thanks for explaining